x-pack/plugins/cross_cluster_replication/public/app/components/auto_follow_pattern_form.js

```diff
@@ -169,16 +169,12 @@ export class AutoFollowPatternForm extends PureComponent {
       this.setState(({ fieldsErrors }) => updateFormErrors(errors, fieldsErrors));
     } else {
       this.setState(({ fieldsErrors, autoFollowPattern: { leaderIndexPatterns } }) => {
-        let errors;
-        if (!leaderIndexPatterns.length) {
-          // If we don't have yet any pattern in our state,
-          // we validate the *current value* of the auto-follow pattern form input
-          errors = validateAutoFollowPattern({ leaderIndexPatterns: [leaderIndexPattern] });
-        } else {
-          // If we do have some auto-follow pattern in our state,
-          // we validate the *value in the state*
-          errors = validateAutoFollowPattern({ leaderIndexPatterns });
-        }
+        // If have at least 1 auto-follow pattern in our state, we validate its value
+        // otherwise, we validate the *current value* of the form input
```

Comment:
Just to make a note of what we talked about over Zoom, I think it would be helpful to adjust this comment to explain the UX intention behind the code, e.g.

```js
const errors = Boolean(leaderIndexPatterns.length)
  // Validate existing patterns, so we can surface an error if this required input is missing.
  ? validateAutoFollowPattern({ leaderIndexPatterns })
  // Validate the input as the user types so they have immediate feedback about errors.
  : validateAutoFollowPattern({ leaderIndexPatterns: [leaderIndexPattern] });
```